nodejs/FatBeaconPeripheral/FatBeacon.js

```diff
@@ -0,0 +1,54 @@
+var bleno = require('bleno');
+var eddystoneBeacon = require('eddystone-beacon');
+
+/***********************Altering Library Functions**************************/
+/**
+ * To make this implementation compatible with existing nodjs libraries,
+ * we employ monkey patching to override functions at runtime to alter
+ * adveristing behavior for Fatbeacon and optimize data transfer rate 
+ * parameters as well.
+ */
+var AdvertisementData = 
+        require('eddystone-beacon/lib/util/advertisement-data');
+var Eir = require('eddystone-beacon/lib/util/eir');
+var FAT_BEACON_FRAME_TYPE = 0x0e;
+
+AdvertisementData.makeEirData = function(serviceData) {
+  var SERVICE_UUID = 'feaa';
+
+  var eir = new Eir();
+  eir.addFlags(0x06);
+  eir.add16BitCompleteServiceList([SERVICE_UUID]);
+  eir.addServiceData(SERVICE_UUID, serviceData);
+
+  return eir.buffer();
+}
+
+AdvertisementData.makeUrlBuffer = function(name) {
+  console.log('AdvertisementData.makeUrlBuffer Called with ' + name);
+
+  var MAX_URL_LENGTH = 18;
+  var FAT_BEACON_FRAME_TYPE = 0x0e;
+  
+  var encodedName = Buffer.from(name);
+  if (encodedName.length > MAX_URL_LENGTH) {
+    throw new Error(`Encoded Name must be less than ${MAX_URL_LENGTH} bytes.` +
+                    ` It is currently ${encodedName.length} bytes.`);
+  }
+
+  var fatBeaconHeader = new Buffer(3);
```

Comment:
can you just put these values in the serviceData concat?  You can use a comment to mark it as the fatbeaconheader